color/src/colorspace.rs

```diff
@@ -0,0 +1,287 @@
+// Copyright 2024 the Color Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use core::f32;
+
+use crate::{matmul, tagged::ColorspaceTag};
+
+#[cfg(all(not(feature = "std"), not(test)))]
+use crate::floatfuncs::FloatFuncs;
+
+/// The main trait for colorspaces.
+///
+/// This can be implemented by clients for conversions in and out of
+/// new colorspaces. It is expected to be a zero-sized type.
+///
+/// The linear sRGB colorspace is central, and other colorspaces are
+/// defined as conversions in and out of that. A colorspace does not
+/// explicitly define a gamut, so generally conversions will succeed
+/// and round-trip, subject to numerical precision.
+///
+/// White point is not explicitly represented. For color spaces with a
+/// white point other than D65 (the native white point for sRGB), use
+/// a linear Bradford chromatic adaptation, following CSS Color 4.
+pub trait Colorspace: Clone + Copy + 'static {
+    /// Whether the colorspace is linear.
+    ///
+    /// Calculations in linear colorspaces can sometimes be simplified,
+    /// for example it is not necessary to undo premultiplication when
+    /// converting.
+    const IS_LINEAR: bool = false;
```

Comment:
Is having defaults here good?

Reply:
For `IS_LINEAR`, yes, for two reasons. First, I find myself motivated by code size, and most colorspaces are not linear. Second, it is safe; if people forget to set it, the only thing that goes wrong is a bit of extra conversion code.

For `CS_TAG`, yes, because it will always be `None` in client impls.

The argument for `LAYOUT` is less compelling, but if we already have defaults I think it makes sense to go with the flow.